src/main/scala/io/pravega/connectors/spark/PravegaBatchPartitionReader.scala

```diff
@@ -52,13 +53,21 @@ case class PravegaBatchPartitionReader(
   override def next(): Boolean = {
     if (iterator.hasNext) {
       val offset = iterator.getOffset
-      val event = iterator.next()
-      nextRow = converter.toUnsafeRow(
-        event.array,
-        segmentRange.getScope,
-        segmentRange.getStreamName,
-        segmentRange.getSegmentId,
-        offset)
+
+      try
+      {
+        val event = iterator.next()
+        nextRow = converter.toUnsafeRow(
+          event.array,
+          segmentRange.getScope,
+          segmentRange.getStreamName,
+          segmentRange.getSegmentId,
+          offset)
+      }
+      catch
+      {
+        case e: TruncatedDataException => log.trace("next: TruncatedDataException while reading data", e)
```

Comment:
In case  iterator.next() throws TruncatedDataException when actual segment has need truncated as part of retention policy set on pravega side. 
Continuing forward without breaking the flow so that reader will not stop and once it reaches to point of truncation iterator will start returning the result.
This is one approach.

Reply:
@crazyzhou  Is there any spark side api that we can call to notify spark about TruncatedDataException.

Reply:
This approach will flood the log with large duplicate lines of `TruncatedDataException`. I think it should simply log once and stop this batch by not providing the next event, let Spark to continue on the next batch.

Reply:
log level changed to warn and returned false so that spark will continue .